directory/directory_transport.go

```diff
@@ -58,6 +58,12 @@ type dirReference struct {
 	// (But in general, we make no attempt to be completely safe against concurrent hostile filesystem modifications.)
 	path         string // As specified by the user. May be relative, contain symlinks, etc.
 	resolvedPath string // Absolute path with no symlinks, at least at the time of its creation. Primarily used for policy namespaces.
+
+	// Suffix options.
+	options []string
+
+	// Does the directory want compressed blobs?
+	compressed bool
```

Comment:
(review remarks on `dir:` just so that they are not lost if we decided to go this way, not as an endorsement of the concept)

Why have both, and silently accept any invalid input? Merely the `compressed` bool would allow us to both parse and format the cases we care about.